# BIG DATA ASSIGNMENT
## 1. Reading Files, Cleaning, and Data Management

**Name:** Júlia Monteiro Santos - A01723519
**Date:** 24/08/2026

## Question 1 - Reading and reviewing the dataset
The cars.csv dataset will be stored in  mis_datos. I will review the structure and basic characteristics of the dataset.

In [1]:
import pandas as pd
import numpy as np

mis_datos = pd.read_csv("../data/cars.csv")

In [2]:
mis_datos.head()

,Make,Model,Type,Origin,DriveTrain,MSRP,Invoice,EngineSize,Cylinders,Horsepower,MPG_City,MPG_Highway,Weight,Wheelbase,Length
0,Acura,MDX,SUV,Asia,All,36945,33337,3.5,6.0,265.0,17.0,23.0,4451.0,106.0,189.0
1,Acura,RSX Type S 2dr,Sedan,Asia,Front,23820,21761,2.0,4.0,200.0,24.0,31.0,2778.0,101.0,172.0
2,Acura,TSX 4dr,Sedan,Asia,Front,26990,24647,2.4,4.0,200.0,22.0,29.0,3230.0,105.0,183.0
3,Acura,TL 4dr,Sedan,Asia,Front,33195,30299,3.2,6.0,270.0,20.0,28.0,NaN,108.0,186.0
4,Acura,3.5 RL 4dr,Sedan,Asia,Front,43755,39014,3.5,6.0,225.0,18.0,24.0,3880.0,115.0,197.0


In [3]:
mis_datos.tail()

,Make,Model,Type,Origin,DriveTrain,MSRP,Invoice,EngineSize,Cylinders,Horsepower,MPG_City,MPG_Highway,Weight,Wheelbase,Length
423,Volvo,C70 LPT convertible 2dr,Sedan,Europe,Front,40565,38203,2.4,5.0,197.0,21.0,28.0,3450.0,105.0,186.0
424,Volvo,C70 HPT convertible 2dr,Sedan,Europe,Front,42565,40083,2.3,5.0,242.0,20.0,26.0,3450.0,105.0,186.0
425,Volvo,S80 T6 4dr,Sedan,Europe,Front,45210,42573,2.9,6.0,268.0,19.0,26.0,3653.0,110.0,190.0
426,Volvo,V40,Wagon,Europe,Front,26135,24641,1.9,4.0,170.0,22.0,29.0,2822.0,101.0,180.0
427,Volvo,XC70,Wagon,Europe,All,35145,33112,2.5,5.0,208.0,20.0,27.0,3823.0,109.0,186.0


In [4]:
mis_datos.sample(5)

,Make,Model,Type,Origin,DriveTrain,MSRP,Invoice,EngineSize,Cylinders,Horsepower,MPG_City,MPG_Highway,Weight,Wheelbase,Length
185,Infiniti,FX35,Wagon,Asia,Rear,34895,31756,3.5,6.0,NaN,16.0,22.0,4056.0,112.0,189.0
5,Acura,3.5 RL w/Navigation 4dr,Sedan,Asia,Front,46100,41100,3.5,6.0,225.0,18.0,24.0,3893.0,115.0,197.0
223,Lexus,IS 300 4dr auto,Sedan,Asia,Rear,32415,28611,3.0,6.0,215.0,18.0,24.0,3285.0,105.0,177.0
120,Ford,Explorer XLT V6,SUV,USA,All,29670,26983,4.0,6.0,210.0,15.0,20.0,4463.0,114.0,190.0
166,Hummer,H2,SUV,USA,All,49995,45815,6.0,8.0,316.0,10.0,12.0,6400.0,123.0,190.0


In [5]:
mis_datos.shape

(428, 15)

In [6]:
mis_datos.columns

Index(['Make', 'Model', 'Type', 'Origin', 'DriveTrain', 'MSRP', 'Invoice',
       'EngineSize', 'Cylinders', 'Horsepower', 'MPG_City', 'MPG_Highway',
       'Weight', 'Wheelbase', 'Length'],
      dtype='object')

In [7]:
mis_datos.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 428 entries, 0 to 427
Data columns (total 15 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   Make         428 non-null    object 
 1   Model        428 non-null    object 
 2   Type         428 non-null    object 
 3   Origin       428 non-null    object 
 4   DriveTrain   428 non-null    object 
 5   MSRP         428 non-null    int64  
 6   Invoice      428 non-null    int64  
 7   EngineSize   425 non-null    float64
 8   Cylinders    425 non-null    float64
 9   Horsepower   423 non-null    float64
 10  MPG_City     426 non-null    float64
 11  MPG_Highway  426 non-null    float64
 12  Weight       423 non-null    float64
 13  Wheelbase    426 non-null    float64
 14  Length       427 non-null    float64
dtypes: float64(8), int64(2), object(5)
memory usage: 50.3+ KB


### Interpretation
It has428 cars and 15 variables. It includes information about the cars such as brand, price, MPG, weight, and other characteristics. Some numerical variables also had missing values.

## Question 2 - Missing values

I will check which variables contain missing values. For the numerical variables with missing info I will replace the missing values with the mean.

In [8]:
mis_datos.isna().sum()

Make           0
Model          0
Type           0
Origin         0
DriveTrain     0
MSRP           0
Invoice        0
EngineSize     3
Cylinders      3
Horsepower     5
MPG_City       2
MPG_Highway    2
Weight         5
Wheelbase      2
Length         1
dtype: int64

In [9]:
columns_with_missing = [
    "EngineSize",
    "Cylinders",
    "Horsepower",
    "MPG_City",
    "MPG_Highway",
    "Weight",
    "Wheelbase",
    "Length"
]

for column in columns_with_missing:
    average = mis_datos[column].mean()
    mis_datos[column] = mis_datos[column].fillna(average)

In [10]:
mis_datos.isna().sum()

Make           0
Model          0
Type           0
Origin         0
DriveTrain     0
MSRP           0
Invoice        0
EngineSize     0
Cylinders      0
Horsepower     0
MPG_City       0
MPG_Highway    0
Weight         0
Wheelbase      0
Length         0
dtype: int64

### Interpretation
There were missing values in 8 numerical variables. I replaced the missing values with the mean of each column, and after that there were no missing values.

## Question 3 - Vehicles by manufacturer

I will count the number of vehicles for each manufacturer and identify the manufacturers with the largest and smallest number of vehicles.

In [11]:
make_counts = mis_datos["Make"].value_counts()

make_counts

Make
Toyota           28
Chevrolet        27
Mercedes-Benz    26
Ford             23
BMW              20
Audi             19
Nissan           17
Honda            17
Chrysler         15
Volkswagen       15
Mitsubishi       13
Dodge            13
Volvo            12
Hyundai          12
Jaguar           12
Subaru           11
Kia              11
Pontiac          11
Mazda            11
Lexus            11
Buick             9
Mercury           9
Lincoln           9
Cadillac          8
GMC               8
Saturn            8
Suzuki            8
Infiniti          8
Acura             7
Saab              7
Porsche           7
Land Rover        3
Oldsmobile        3
Jeep              3
Isuzu             2
MINI              2
Scion             2
Hummer            1
Name: count, dtype: int64

In [12]:
largest_count = make_counts.max()
smallest_count = make_counts.min()

print("Largest number of vehicles:", largest_count)
print(make_counts[make_counts == largest_count])

print("\nSmallest number of vehicles:", smallest_count)
print(make_counts[make_counts == smallest_count])

Largest number of vehicles: 28
Make
Toyota    28
Name: count, dtype: int64

Smallest number of vehicles: 1
Make
Hummer    1
Name: count, dtype: int64


### Interpretation
Toyota has the most cars in the dataset with 28 and Hummer has the least with only 1 vehicle.

## Question 4 - Vehicles by origin and drivetrain

I will group the vehicles by origin and drivetrain to determine how many vehicles belong to each combination.

In [13]:
vehicles_by_region = (
    mis_datos
    .groupby(["Origin", "DriveTrain"])
    .size()
)

vehicles_by_region

Origin  DriveTrain
Asia    All           34
        Front         99
        Rear          25
Europe  All           36
        Front         37
        Rear          50
USA     All           22
        Front         90
        Rear          35
dtype: int64

### Interpretation
Asia has the most Front cars with 99. Europe has the most Rear cars with 50 and the most All-wheel-drive cars with 36.

## Question 5 - Price and fuel economy summary

I will calculate the highest and lowest suggested retail prices, alsp the average city and highway miles per gallon.

In [14]:
highest_msrp = mis_datos["MSRP"].max()
lowest_msrp = mis_datos["MSRP"].min()
average_city_mpg = mis_datos["MPG_City"].mean()
average_highway_mpg = mis_datos["MPG_Highway"].mean()

print("Highest MSRP:", highest_msrp)
print("Lowest MSRP:", lowest_msrp)
print("Average city MPG:", average_city_mpg)
print("Average highway MPG:", average_highway_mpg)

Highest MSRP: 192465
Lowest MSRP: 10280
Average city MPG: 20.065727699530516
Average highway MPG: 26.840375586854464


### Interpretation
The highest MSRP is $192,465 and the lowest is $10,280. The average city MPG is about 20.07 while the average highway MPG is about 26.84.

## Question 6 - Selection and filtering

I will select only required variables and keep the observations where DriveTrain is same to Front.

In [15]:
mis_nuevos_datos = mis_datos[
    [
        "Make",
        "Model",
        "Type",
        "Origin",
        "DriveTrain",
        "MSRP",
        "Weight",
        "Wheelbase",
        "Length"
    ]
].copy()

In [16]:
mis_nuevos_datos = mis_nuevos_datos[
    mis_nuevos_datos["DriveTrain"] == "Front"
].copy()

mis_nuevos_datos.head()

,Make,Model,Type,Origin,DriveTrain,MSRP,Weight,Wheelbase,Length
1,Acura,RSX Type S 2dr,Sedan,Asia,Front,23820,2778.000000,101.0,172.0
2,Acura,TSX 4dr,Sedan,Asia,Front,26990,3230.000000,105.0,183.0
3,Acura,TL 4dr,Sedan,Asia,Front,33195,3578.229314,108.0,186.0
4,Acura,3.5 RL 4dr,Sedan,Asia,Front,43755,3880.000000,115.0,197.0
5,Acura,3.5 RL w/Navigation 4dr,Sedan,Asia,Front,46100,3893.000000,115.0,197.0


### Interpretation
I selected only the variables required in the assignment and filtered the data to keep only cars with Front drivetrain.

## Question 7 - Creation and transformation of variables

I will display the first observations of misnuevosdatos and create 3 new variables by converting weight from pounds to kilograms andthe  Wheelbase and Length from inches to centimeters.

In [17]:
mis_nuevos_datos.head()

,Make,Model,Type,Origin,DriveTrain,MSRP,Weight,Wheelbase,Length
1,Acura,RSX Type S 2dr,Sedan,Asia,Front,23820,2778.000000,101.0,172.0
2,Acura,TSX 4dr,Sedan,Asia,Front,26990,3230.000000,105.0,183.0
3,Acura,TL 4dr,Sedan,Asia,Front,33195,3578.229314,108.0,186.0
4,Acura,3.5 RL 4dr,Sedan,Asia,Front,43755,3880.000000,115.0,197.0
5,Acura,3.5 RL w/Navigation 4dr,Sedan,Asia,Front,46100,3893.000000,115.0,197.0


In [18]:
mis_nuevos_datos["Weight_kilo"] = mis_nuevos_datos["Weight"] * 0.453592
mis_nuevos_datos["Wheelbase_cm"] = mis_nuevos_datos["Wheelbase"] * 2.54
mis_nuevos_datos["Length_cm"] = mis_nuevos_datos["Length"] * 2.54

mis_nuevos_datos.head()

,Make,Model,Type,Origin,DriveTrain,MSRP,Weight,Wheelbase,Length,Weight_kilo,Wheelbase_cm,Length_cm
1,Acura,RSX Type S 2dr,Sedan,Asia,Front,23820,2778.000000,101.0,172.0,1260.078576,256.54,436.88
2,Acura,TSX 4dr,Sedan,Asia,Front,26990,3230.000000,105.0,183.0,1465.102160,266.70,464.82
3,Acura,TL 4dr,Sedan,Asia,Front,33195,3578.229314,108.0,186.0,1623.056191,274.32,472.44
4,Acura,3.5 RL 4dr,Sedan,Asia,Front,43755,3880.000000,115.0,197.0,1759.936960,292.10,500.38
5,Acura,3.5 RL w/Navigation 4dr,Sedan,Asia,Front,46100,3893.000000,115.0,197.0,1765.833656,292.10,500.38


### Interpretation
I created three new variables to show the weight in kilograms and the Wheelbase and Length in centimeters.

## Question 8 - Price classification

I will create a new variable called opinion based on the suggested retail price of each vehicle.

In [19]:
def classify_price(price):
    if price > 60000:
        return "Expensive"
    elif price < 30000:
        return "Inexpensive"
    else:
        return "Normal"

In [20]:
mis_nuevos_datos["opinion"] = mis_nuevos_datos["MSRP"].apply(classify_price)

mis_nuevos_datos.head()

,Make,Model,Type,Origin,DriveTrain,MSRP,Weight,Wheelbase,Length,Weight_kilo,Wheelbase_cm,Length_cm,opinion
1,Acura,RSX Type S 2dr,Sedan,Asia,Front,23820,2778.000000,101.0,172.0,1260.078576,256.54,436.88,Inexpensive
2,Acura,TSX 4dr,Sedan,Asia,Front,26990,3230.000000,105.0,183.0,1465.102160,266.70,464.82,Inexpensive
3,Acura,TL 4dr,Sedan,Asia,Front,33195,3578.229314,108.0,186.0,1623.056191,274.32,472.44,Normal
4,Acura,3.5 RL 4dr,Sedan,Asia,Front,43755,3880.000000,115.0,197.0,1759.936960,292.10,500.38,Normal
5,Acura,3.5 RL w/Navigation 4dr,Sedan,Asia,Front,46100,3893.000000,115.0,197.0,1765.833656,292.10,500.38,Normal


### Interpretation
The opinion variable marks the cars based on their MSRP. Cars less than $30,000 are Inexpensive, cars $30,000 to $60,000 are Normal and the cars higher than $60,000 are Expensive.